database/intializeDB.go

```diff
@@ -3,49 +3,146 @@ package database
 import (
 	"database/sql"
 	"fmt"
+	"net/http"
+	"strconv"
+	"time"
 
 	_ "github.com/lib/pq"
 
-	"github.com/LetsFocus/goLF/configs"
 	"github.com/LetsFocus/goLF/errors"
+	"github.com/LetsFocus/goLF/goLF/model"
+	"github.com/LetsFocus/goLF/logger"
 )
 
-type config struct {
-	host     string
-	password string
-	user     string
-	port     string
-	dialect  string
-	dbName   string
+type dbConfig struct {
+	host                  string
+	password              string
+	user                  string
+	port                  string
+	dialect               string
+	dbName                string
+	maxOpenConns          int
+	maxIdleConns          int
+	connMaxLifeTime       int
+	idleConnectionTimeout int
+	monitoringEnable      bool
+	retry                 int
+	retryDuration         int
 }
 
-func InitializeDB(configs configs.Config, prefix string) (*sql.DB, error) {
-	c := config{host: configs.Get(prefix + "DB_HOST"), password: configs.Get(prefix + "DB_PASSWORD"),
-		user: configs.Get(prefix + "DB_USER"), port: configs.Get(prefix + "DB_PORT"), dialect: configs.Get(prefix + "DB_DIALECT"),
-		dbName: configs.Get(prefix + "DB_NAME")}
+func InitializeDB(golf model.GoLF, prefix string) {
+	var (
+		maxConnections, maxIdleConnections, connectionMaxLifeTime, idleConnectionTimeout, retry, retryTime int
+		monitoring                                                                                         bool
+		err                                                                                                error
+	)
 
-	connectionString := generateConnectionString(c)
-	if connectionString == "" {
-		return nil, errors.Errors{StatusCode: 500, Code: "Invalid Dialect", Reason: "invalid dialect given"}
+	maxIdleConnections, err = strconv.Atoi(golf.Config.Get(prefix + "DB_MAX_IDLE_CONNECTIONS"))
+	if err != nil {
+		maxIdleConnections = 5
 	}
 
-	db, err := sql.Open(c.dialect, connectionString)
+	maxConnections, err = strconv.Atoi(golf.Config.Get(prefix + "DB_MAX_CONNECTIONS"))
 	if err != nil {
-		configs.Log.Errorf("Failed to initialize the DB, Error:%v", err)
-		return nil, err
+		maxConnections = 20
 	}
 
-	err = db.Ping()
+	connectionMaxLifeTime, err = strconv.Atoi(golf.Config.Get(prefix + "DB_CONNECTIONS_MAX_LIFETIME"))
 	if err != nil {
-		configs.Log.Errorf("Failed to initialize the DB, Error:%v", err)
-		return nil, err
+		connectionMaxLifeTime = 15
 	}
 
-	configs.Log.Info("database is connected successfully")
-	return db, nil
+	idleConnectionTimeout, err = strconv.Atoi(golf.Config.Get(prefix + "DB_IDLE_CONNECTION_TIMEOUT"))
+	if err != nil {
+		idleConnectionTimeout = 10
+	}
+
+	monitoring, err = strconv.ParseBool(golf.Config.Get(prefix + "DB_MONITORING"))
+	if err != nil {
+		monitoring = false
+	}
+
+	retry, err = strconv.Atoi(golf.Config.Get(prefix + "DB_RETRY_COUNT"))
+	if err != nil {
+		retry = 5
+	}
+
+	retryTime, err = strconv.Atoi(golf.Config.Get(prefix + "DB_RETRY_DURATION"))
```

Comment:
What is DB retry duration?

Reply:
It will do retry right basically this is gap between the retries